akka-stream/src/main/scala/akka/stream/Queue.scala

```diff
@@ -12,13 +12,21 @@ trait SourceQueue[T] {
 
   /**
    * Method offers next element to a stream and returns future that:
-   * - competes with true if element is consumed by a stream
-   * - competes with false when stream dropped offered element
-   * - fails if stream is completed or cancelled.
+   * - competes with `Success(true)` if element is consumed by a stream
+   * - competes with `Success(false)` when stream dropped offered element
+   * - competes with `StreamCompleted` when stream is completed during future is active
+   * - competes with `Failure(f)` when stream is failed
```

Comment:
changed this to "when host stage is failed". queue can fail when OverflowStrategy is Fail